src/js/App/GlobalFilter/TagsModal.tsx

```diff
@@ -7,126 +7,132 @@ import debounce from 'lodash/debounce';
     return [selected?.isLoaded, selected?.total || 0, selected?.page || 1, selected?.perPage || 10];
   }, shallowEqual);
 
-const TagsModal = ({ isOpen, filterTagsBy, onApplyTags, toggleModal, selectedTags }: TagsModalProps) => {
-  const intl = useIntl();
-  const [tagsSelected, setTagsSelected] = useState<GlobalFilterTag[]>([]);
-  const [sidsSelected, setSidsSelected] = useState<SID[]>([]);
-  const [filterBy, setFilterBy] = useState('');
-  const [filterSIDsBy, setFilterSIDsBy] = useState('');
-  const dispatch = useDispatch();
-  const [tagsLoaded, tagsCount, tagsPage, tagsPerPage] = useMetaSelector('tags');
-  const [sidLoaded, sidCount, sidPage, sidPerPage] = useMetaSelector('sid');
-  const tags = useSelector<ReduxState, GlobalFilterTag[]>(({ globalFilter: { tags } }) => tags?.items || []);
-  const sids = useSelector<ReduxState, SID[]>(({ globalFilter: { sid } }) => sid?.items || []);
-  const filterScope = useSelector<ReduxState, 'insights' | undefined>(({ globalFilter: { scope } }) => scope || undefined);
-  const debounceGetTags = useCallback(
-    debounce((search?: string) => {
-      dispatch(
-        fetchAllTags(
-          {
-            registeredWith: filterScope,
-            activeTags: selectedTags,
-            search,
-          },
-          { page: tagsPage, perPage: tagsPerPage }
-        )
+const usePagination = (loaded: boolean | unknown, perPage?: number, page?: number, count?: number) => {
+  return useMemo(() => {
+    if (loaded) {
+      return {
+        perPage: perPage,
+        page: page,
+        count: count,
+      };
+    }
+    return {};
+  }, [loaded, perPage, page, count]);
+};
+
+const useRow = (
+  resource: SID[] | GlobalFilterTag[],
+  loaded: boolean | unknown,
+  idMapper: IDMapper,
+  cellsMapper: CellsMapper,
+  selected?: GlobalFilterTag[]
+) => {
+  return useMemo(() => {
+    if (loaded) {
+      return flatMap(resource, ({ tags }) =>
+        tags?.map(({ tag }) => ({
+          id: idMapper(tag),
+          namespace: tag.namespace,
+          key: tag.key,
+          value: tag.value,
+          selected: selected?.find?.(({ id }) => id === idMapper(tag)),
+          cells: cellsMapper(tag),
+        }))
       );
-    }, 800),
-    []
-  );
-  const debounceGetSIDs = useCallback(
+    }
+    return [];
+  }, [resource, selected]);
+};
+
+const useDebounce = (callback: DebounceCallback, perPage: number, activeTags?: FlagTagsFilter) => {
+  const registeredWith = useSelector<ReduxState, 'insights' | undefined>(({ globalFilter: { scope } }) => scope || undefined);
+  const dispatch = useDispatch();
+  return useCallback(
     debounce((search?: string) => {
       dispatch(
-        fetchAllSIDs(
+        callback(
           {
-            registeredWith: filterScope,
-            activeTags: selectedTags,
+            registeredWith,
+            activeTags,
             search,
           },
-          { page: sidPage, perPage: sidPerPage }
+          { page: 1, perPage }
```

Comment:
Is the page filter always 1?

Reply:
It can be something else, but given the problem with inventory throwing 404 if no data are found for said page I think it's safer to always return to first page.

Reply:
Ah yeah that bit.